include/fuse.h

```diff
@@ -750,6 +750,15 @@ struct fuse_operations {
 	 */
 	int (*fallocate) (const char *, int, off_t, off_t,
 			  struct fuse_file_info *);
+
+	/**
+	 * Copy a range of data from one file to another
+	 *
+	 * Performs an optimized copy between two file descriptors without the
+	 * additional cost of transferring data through the FUSE kernel module
+	 * to user space (glibc) and then back into the FUSE filesystem again.
+	 */
+	ssize_t (*copy_file_range) (int, off_t, int, off_t, size_t, int);
```

Comment:
This definition is incorrect – FUSE operations which operate on file descriptors take `fuse_file_info* fi` arguments, not `int fd`. Additionally, this declaration doesn't follow conventions used in the FUSE's high-level interface where callbacks which operate on file descriptors are also given paths to the files. Thus, this should rather be declared as:
```c
ssize_t (*copy_file_range) (const char* path_in, fuse_file_info* fi_in, off_t offset_in,
                            const char* path_out, fuse_file_info* fi_out, off_t offset_out,
                            size_t size, int flags);
```

Reply:
Ok, if that is the convention, I'll change it.